libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -259,32 +259,60 @@ ArraySchema ArrowAdapter::tiledb_schema_from_arrow_schema(
         if (platform_config->offsets_filters.size() != 0) {
             FilterList offset_filter_list(*ctx);
             for (auto offset : platform_config->offsets_filters) {
-                offset_filter_list.add_filter(
-                    Filter(*ctx, convert_filter[offset]));
+                try {
+                    offset_filter_list.add_filter(
+                        Filter(*ctx, convert_filter.at(offset)));
+                } catch (std::out_of_range& e) {
+                    throw TileDBSOMAError(fmt::format(
+                        "Invalid offset filter {} passed to PlatformConfig",
+                        offset));
+                }
             }
             schema.set_offsets_filter_list(offset_filter_list);
         }
 
         if (platform_config->validity_filters.size() != 0) {
             FilterList validity_filter_list(*ctx);
             for (auto validity : platform_config->validity_filters) {
-                validity_filter_list.add_filter(
-                    Filter(*ctx, convert_filter[validity]));
+                try {
+                    validity_filter_list.add_filter(
+                        Filter(*ctx, convert_filter.at(validity)));
+                } catch (std::out_of_range& e) {
+                    throw TileDBSOMAError(fmt::format(
+                        "Invalid validity filter {} passed to PlatformConfig",
+                        validity));
+                }
             }
             schema.set_validity_filter_list(validity_filter_list);
         }
 
         schema.set_allows_dups(platform_config->allows_duplicates);
 
-        if (platform_config->tile_order)
-            schema.set_tile_order(
-                platform_config->tile_order == "row" ? TILEDB_ROW_MAJOR :
-                                                       TILEDB_COL_MAJOR);
+        if (platform_config->tile_order) {
+            if (platform_config->tile_order == "row") {
```

Comment:
I am wondering if we could extend the comparison. I get passed "ROW_ORDER" and "COL_ORDER", respectively, from @mojaveazure

Reply:
There is a check for both "row" and "col" here, so I'm not sure what you mean.

Reply:
To add an `|| platform_config->tile_order == "ROW_ORDER"` as that is the value I (as the middle man) receive.  Saves me from mapping it.

Do we what if anything the spec suggests?  I suppose you got `row` from Python. If R uses `ROW_ORDER` then one of the two may be wrong...

Reply:
OIC. Yeah I can modify this.

Reply:
The R package currently has these strings as defaults:

```r
.default_tile_order        <- function() { "ROW_MAJOR" }
.default_cell_order        <- function() { "ROW_MAJOR" }

# ...

.default_offsets_filters  <- function() { list("DOUBLE_DELTA", "BIT_WIDTH_REDUCTION", "ZSTD")}
```

There is a bit of a larger issue looming that maybe we want the C++ side to own the defaults and have the API just instantiate the difference from defaults.  Will try to chat with @mojaveazure about his side of things...